src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1179,6 +1265,208 @@ static void uct_ib_mlx5_devx_md_cleanup(uct_ib_md_t *ibmd)
+    void *hdr       = UCT_IB_MLX5DV_ADDR_OF(create_alias_obj_in, in, hdr);
+    void *alias_ctx = UCT_IB_MLX5DV_ADDR_OF(create_alias_obj_in, in,
+                                            alias_ctx);
+    void *access_key;
+    int rc;
+
+    /* obtain pdn */
+    dv.pd.in  = md->super.pd;
+    dv.pd.out = &dvpd;
+    rc        = mlx5dv_init_obj(&dv, MLX5DV_OBJ_PD);
+    if (rc) {
+        uct_md_log_mem_attach_error(flags, "mlx5dv_init_obj() failed: %m");
+        return UCS_ERR_IO_ERROR;
+    }
+
+    /* create alias */
+    UCT_IB_MLX5DV_SET(general_obj_in_cmd_hdr, hdr, opcode,
+                      UCT_IB_MLX5_CMD_OP_CREATE_GENERAL_OBJECT);
+    UCT_IB_MLX5DV_SET(general_obj_in_cmd_hdr, hdr, obj_type,
+                      UCT_IB_MLX5_OBJ_TYPE_MKEY);
+    UCT_IB_MLX5DV_SET(general_obj_in_cmd_hdr, hdr, alias_object, 1);
+    UCT_IB_MLX5DV_SET(alias_context, alias_ctx, vhca_id_to_be_accessed,
+                      target_gvmi_id);
+    UCT_IB_MLX5DV_SET(alias_context, alias_ctx, object_id_to_be_accessed,
+                      target_mkey >> 8);
+    UCT_IB_MLX5DV_SET(alias_context, alias_ctx, metadata_1, dvpd.pdn);
+    access_key = UCT_IB_MLX5DV_ADDR_OF(alias_context, alias_ctx, access_key);
+    ucs_strncpy_zero(access_key, uct_ib_mkey_token,
+                     UCT_IB_MLX5DV_FLD_SZ_BYTES(alias_context, access_key));
+
+    memh->cross_mr = uct_ib_mlx5_devx_obj_create(md->super.dev.ibv_context, in,
+                                                 sizeof(in), out, sizeof(out),
+                                                 "MKEY_ALIAS",
+                                                 uct_md_attach_log_lvl(flags));
+    if (memh->cross_mr == NULL) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    rc = UCT_IB_MLX5DV_GET(create_alias_obj_out, out, alias_ctx.status);
+    if (rc) {
+        uct_md_log_mem_attach_error(flags,
+                                    "created MR alias object in bad state,"
+                                    " syndrome 0x%x",
+                                    UCT_IB_MLX5DV_GET(create_alias_obj_out,
+                                                      out, hdr.syndrome));
+        return UCS_ERR_IO_ERROR;
+    }
+
+    memh->super.lkey = UCT_IB_MLX5DV_GET(create_alias_obj_out, out, hdr.obj_id)
+                       << 8;
+    memh->super.rkey = memh->super.lkey;
+
+    return UCS_OK;
+}
+
+#else
+
+static ucs_status_t
+uct_ib_mlx5_devx_reg_exported_key(uct_ib_md_t *ib_md, uint64_t flags,
+                                  void *address, size_t length,
+                                  uct_ib_mem_t *ib_memh)
+{
+    return UCS_ERR_UNSUPPORTED;
+}
+
+static ucs_status_t
+uct_ib_mlx5_devx_import_exported_key(uct_ib_md_t *ib_md, uint64_t flags,
+                                     uint32_t target_gvmi_id,
+                                     uint32_t target_mkey,
+                                     uct_ib_mem_t *ib_memh)
+{
+    return UCS_ERR_UNSUPPORTED;
+}
```

Comment:
IMO this does not have to be under ifdef

Reply:
uct_ib_mlx5_devx_import_exported_key does not call umem_reg_ex so it can be outside of `#if`